diff --git a/cpp/rnskia/dom/base/DerivedNodeProp.h b/cpp/rnskia/dom/base/DerivedNodeProp.h
index 64b817edab09d44c8b7c53a4d083020092aa7c7a..3a9ecb4476b1b53fdfdb1bc4d3b0f95ae44616ca 100644
--- a/cpp/rnskia/dom/base/DerivedNodeProp.h
+++ b/cpp/rnskia/dom/base/DerivedNodeProp.h
@@ -72,7 +72,7 @@ public:
    Adds a property to the derived property child props.
    */
   template <class _Tp, class... _Args,
-            class = std::_EnableIf<!std::is_array<_Tp>::value>>
+            class = std::enable_if_t<!std::is_array<_Tp>::value>>
   _Tp *defineProperty(_Args &&...__args) {
     auto prop =
         std::make_shared<_Tp>(std::forward<_Args>(__args)..., _onChange);
diff --git a/cpp/rnskia/dom/base/NodePropsContainer.h b/cpp/rnskia/dom/base/NodePropsContainer.h
index 380764ccdf02f00c85f06647284417d1d0918594..10f1ac982c751bc8abf5f111d0612cbc7986601a 100644
--- a/cpp/rnskia/dom/base/NodePropsContainer.h
+++ b/cpp/rnskia/dom/base/NodePropsContainer.h
@@ -106,7 +106,7 @@ public:
    Defines a property that will be added to the container
    */
   template <class _Tp, class... _Args,
-            class = std::_EnableIf<!std::is_array<_Tp>::value>>
+            class = std::enable_if_t<!std::is_array<_Tp>::value>>
   _Tp *defineProperty(_Args &&...__args) {
     // Create property and set onChange callback
     auto prop =
