diff --git a/ios/CameraError.swift b/ios/CameraError.swift
index 056bf13a7f3e9682fb408cafd4ac4062f2218627..7623cf3fefd2c23989e1a1ad7177e93db03747e8 100644
--- a/ios/CameraError.swift
+++ b/ios/CameraError.swift
@@ -261,7 +261,6 @@ enum CameraError: Error {
   case format(_ id: FormatError)
   case session(_ id: SessionError)
   case capture(_ id: CaptureError)
-  case system(_ id: SystemError)
   case unknown(message: String? = nil)
 
   var code: String {
diff --git a/src/CameraDevice.ts b/src/CameraDevice.ts
index 2e2c76633e769a5df019e61898aaf80c5b4ae1da..6d6780373671bed98d2f64fb2b0bc352d54969bf 100644
--- a/src/CameraDevice.ts
+++ b/src/CameraDevice.ts
@@ -27,7 +27,6 @@ export const parsePhysicalDeviceTypes = (
   physicalDeviceTypes: PhysicalCameraDeviceType[],
 ): PhysicalCameraDeviceType | LogicalCameraDeviceType => {
   if (physicalDeviceTypes.length === 1) {
-    // @ts-expect-error for very obvious reasons
     return physicalDeviceTypes[0];
   }
 
